### ISSUE TITLE
Un-ignore tests from coverage

### ISSUE THREAD

**Issue by samuelallan72** [canonical/charmed-openstack-upgrader#455]
It's just as important that the tests are included in coverage.

- identify tests that aren't run (eg. typo in function name, function name shadowing)
- identify branches in tests that aren't run (eg. a parameterised test with if/else conditions may not be running as expected)
- tests should almost always have 100% coverage; anything under signifies dead code

When we do the un-ignoring, we should also do a pass over the coverage to identify and fix such issues.

---

I un-ignored the tests directory and ran `tox -e unit` to give an example of the types of issues that this will identify:

`generate_expected_upgrade_plan_principal()` is not used in the codebase:

![1719450892](https://github.com/canonical/charmed-openstack-upgrader/assets/9714796/2d571a1f-e06c-4648-9555-1408b0d3a2d6)

One of the asserts in `test_parse_args_help()` is not being run (blocked by the exception raised on the previous line):

![1719450952](https://github.com/canonical/charmed-openstack-upgrader/assets/9714796/0b2a3ccf-6a8e-4199-9251-18a28679b999)

There's a line never called on `TestModel` in `test_retry_failure()`, suggesting a refactoring or removal of the line:

![1719451033](https://github.com/canonical/charmed-openstack-upgrader/assets/9714796/47a4447f-a9cd-4def-9491-6271bd7f5b45)

**Comment by Pjack** [comment]
Closing this issue as it is low priority and there are no immediate plans to address it.
